File: src/test/java/aoc2015/day03/Day03Test.java
package aoc2015.day03;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day03Test {

    @Test
    void part1Test() {
        var day = new Day03();

        assertEquals(2, day.part1(">"));
        assertEquals(4, day.part1("^>v<"));
        assertEquals(2, day.part1("^v^v^v^v^v^v"));
    }

    @Test
    void part2Test() {
        var day = new Day03();

        assertEquals(3, day.part2("^v"));
        assertEquals(3, day.part2("^>v<"));
        assertEquals(11, day.part2("^v^v^v^v^v"));
    }
}

File: aoc20/src/test/java/day03/Day03Test.kt
package day03

import aoc15.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day03Test {
    val input: String = loadExample("day03.txt")

    @Test
    fun part1Example() {
        val subject = Day03()

        assertThat(subject.part1(input))
            .isEqualTo(7)
    }

    @Test
    fun part2Example() {
        val subject = Day03()

        assertThat(subject.part2(input))
            .isEqualTo(336)
    }
}